File: public/benchmarks/simd/zig/simd_scalar.zig
const std = @import("std");
const N = 20000000;
var a: [N]f64 = undefined;
pub fn main() void {
    var i: usize = 0;
    while (i < N) : (i += 1) a[i] = @as(f64, @floatFromInt(i % 1000)) * 0.001;
    var sum: f64 = 0.0;
    i = 0;
    while (i < N) : (i += 1) sum += a[i] * a[i];
    std.debug.print("{d:.6}\n", .{sum});
}

File: public/benchmarks/simd/rust/simd_scalar.rs
const N: usize = 20_000_000;
fn main() {
    let mut a = vec![0.0f64; N];
    for i in 0..N {
        a[i] = (i % 1000) as f64 * 0.001;
    }
    let mut sum = 0.0f64;
    for i in 0..N {
        sum += a[i] * a[i];
    }
    println!("{:.6}", sum);
}
